compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -2117,4 +2117,38 @@ object messages {
     val explanation: String = ""
   }
 
+  case class PatternMatchAlwaysSucceeds(foundCls: Symbol, testCls: Symbol)(implicit ctx: Context) extends Message(PatternMatchAlwaysSucceedsID) {
+    val kind = "Syntax"
+    val msg: String =
+      hl"""
+         |The type check in the highlighted part of the pattern match case statement will always succeed,
+         |since `$foundCls` is a subclass of `$testCls`.
+       """
+    val codeExampleValidWarning = "def foo(a: Int) = a match { case x: Int => x }"
+    val codeExampleDisregardWarning =
+      hl"""
+        |object Foo {
+        |  def unapply(x: Int): Option[Int] = if (x % 2 == 0) Some(x) else None
+        |}
+        |
+        |class Test {
+        |  def foo(a: Int) = a match { case Foo(x: Int) => x }
+        |}
+      """
+    val explanation: String =
```

Comment:
I would remove the current explanation. I don't think it gives you more information than the `msg`.